### PR TITLE
Fix typing in `pl.plugins.environments`

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Part of #7037 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda @awaelchli @ananthsub

### CODE REVIEW

File: pytorch_lightning/plugins/environments/lsf_environment.py [hunk 1]
```diff
@@ -148,15 +149,14 @@ def _get_main_port() -> int:
...
+            log.debug(f"using externally specified main port: {os.environ['MASTER_PORT']}")
+            return int(os.environ["MASTER_PORT"])
+        if "LSB_JOBID" in os.environ:
+            port = int(os.environ["LSB_JOBID"])
             # all ports should be in the 10k+ range
-            port = int(port) % 1000 + 10000
+            port = port % 1000 + 10000
             log.debug(f"calculated LSF main port: {port}")
+            return port
         else:
```
Comment (carmocca) [inline pytorch_lightning/plugins/environments/lsf_environment.py:161]: Unnecessary else

File: pytorch_lightning/plugins/environments/lsf_environment.py [hunk 2]
```diff
@@ -148,15 +149,14 @@ def _get_main_port() -> int:
...
         """
         # check for user-specified main port
-        port = os.environ.get("MASTER_PORT")
-        if not port:
-            jobid = os.environ.get("LSB_JOBID")
-            if not jobid:
-                raise ValueError("Could not find job id in environment variable LSB_JOBID")
-            port = int(jobid)
+        if "MASTER_PORT" in os.environ:
+            log.debug(f"using externally specified main port: {os.environ['MASTER_PORT']}")
```
Comment (carmocca) [inline pytorch_lightning/plugins/environments/lsf_environment.py:153]:
```suggestion
            log.debug(f"Using externally specified main port: {os.environ['MASTER_PORT']}")
```

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/plugins/environments/lsf_environment.py [hunk 3]
```diff
@@ -148,15 +149,13 @@ def _get_main_port() -> int:
...
         """
         # check for user-specified main port
-        port = os.environ.get("MASTER_PORT")
-        if not port:
-            jobid = os.environ.get("LSB_JOBID")
-            if not jobid:
-                raise ValueError("Could not find job id in environment variable LSB_JOBID")
-            port = int(jobid)
+        if "MASTER_PORT" in os.environ:
+            log.debug(f"Using externally specified main port: {os.environ['MASTER_PORT']}")
```
Comment (tchaton) [inline pytorch_lightning/plugins/environments/lsf_environment.py:153]: Should we use rank_zero_debug ?

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/plugins/environments/torchelastic_environment.py [hunk 4]
```diff
@@ -55,18 +53,16 @@ def main_port(self) -> int:
...
 
     @staticmethod
     def detect() -> bool:
         """Returns ``True`` if the current process was launched using the torchelastic command."""
         required_env_vars = {"RANK", "GROUP_RANK", "LOCAL_RANK", "LOCAL_WORLD_SIZE"}
         return required_env_vars.issubset(os.environ.keys())
 
-    def world_size(self) -> Optional[int]:
-        world_size = os.environ.get("WORLD_SIZE")
-        return int(world_size) if world_size is not None else world_size
```
Comment (four4fish) [inline pytorch_lightning/plugins/environments/torchelastic_environment.py:69]: Is that possible that word_size be None?

**Comment by four4fish** [comment]
Will this be a breaking change?

File: pytorch_lightning/plugins/environments/torchelastic_environment.py (hunk 4 above)
Comment (ananthsub) [reply pytorch_lightning/plugins/environments/torchelastic_environment.py:69]: No. WORLD_SIZE should always be made available by torch distributed elastic: https://pytorch.org/docs/stable/elastic/run.html

**Comment by awaelchli** [comment]
Not to my knowledge. It's not my intention. I am quite certain that everywhere in this PR where I changed `os.environ.get("X")` to `os.environ["X"]` it was actually never possible to return None.

**Pull request merged**